### PR TITLE
feat: file managment system

### PR DESCRIPTION
**Pull request by Dheebz**
## Summary

Adds basic file management to the file explorer.

You can now do common file operations directly from the explorer instead of bouncing out to command mode:

- Create files and directories (`Alt+a`)
- Delete with confirmation (`Alt+d`)
- Rename in place (`Alt+r`)
- Move files (`Alt+m`)
- Copy/yank to clipboard (`Alt+y`)
- Cut to clipboard (`Alt+x`)
- Paste from clipboard (`Alt+p`)

This is designed to be an very lightweight minimal framework to impliment the solution with the ability for more "aesthetic" improvements over time. 

## Motivation
This change brings the explorer closer to what people expect from tools like `nvim-tree` or `neo-tree`, while keeping everything keyboard-driven and lightweight. Additionally previous PR's associated with have been since been closed by their authors.

## Implementation

The core of this change is a new `FileExplorer` component in  
`helix-term/src/ui/file_explorer.rs`.

It wraps the existing `Picker`, intercepting file-management keybindings and delegating everything else back to the picker as-is.

A few notable details:

- The clipboard is global (`OnceLock<Mutex<…>>`), so yank/cut persists even if you close and reopen the explorer.
- Cut operations automatically clear the clipboard after a successful paste.
- Moves across filesystems fall back to copy-then-delete when `rename()` fails with `EXDEV`.

## Keybindings

| Key | Action | Notes |
|-----|--------|-------|
| `Alt+a` | Create | Trailing `/` creates a directory. Nested paths (e.g. `foo/bar/file.txt`) auto-create parents |
| `Alt+d` | Delete | Prompts `(y/N)` |
| `Alt+r` | Rename | Pre-fills the current name |
| `Alt+m` | Move | Supports `~` expansion |
| `Alt+y` | Yank | Copy to clipboard |
| `Alt+x` | Cut | Move to clipboard (deleted on paste) |
| `Alt+p` | Paste | Copies or moves depending on whether the source was yanked or cut |

## Testing

Added 8 unit tests covering:

- Directory copying
- Clipboard behavior
- Path handling edge cases

Run with:

```bash
cargo test --package helix-term file_explorer
```

---
Resolves: https://github.com/helix-editor/helix/issues/15136

### CODE REVIEW

File: helix-term/src/ui/file_explorer.rs [hunk 1]
```diff
@@ -0,0 +1,1007 @@
...
+                            verb,
+                            path.display(),
+                            dest_path.display()
+                        ));
+                        schedule_refresh_with_prompt(cx, root.clone());
+                    }
+                    Err(e) => {
+                        // Cross-device move: try copy + delete
+                        if e.raw_os_error() == Some(18) {
+                            // EXDEV
```
[the remark this answers is not in the document]
Comment (Dheebz) [reply helix-term/src/ui/file_explorer.rs]: Resolved with https://github.com/helix-editor/helix/pull/15174/commits/ee86bbd96c82ef762e9c75142b93ea93c269672d

File: helix-term/src/ui/file_explorer.rs [hunk 2]
```diff
@@ -0,0 +1,1007 @@
...
+            return EventResult::Consumed(None);
+        }
+
+        if let Ok(mut clipboard) = get_clipboard().lock() {
+            *clipboard = Some(ClipboardEntry {
+                path: path.clone(),
+                operation: ClipboardOperation::Cut,
+            });
+        }
+        ctx.editor.set_status(format!("Cut: {}", path.display()));
```
[the remark this answers is not in the document]
Comment (Dheebz) [reply helix-term/src/ui/file_explorer.rs]: Resolved with https://github.com/helix-editor/helix/pull/15174/commits/ee86bbd96c82ef762e9c75142b93ea93c269672d

File: helix-term/src/ui/file_explorer.rs [hunk 3]
```diff
@@ -0,0 +1,1007 @@
...
+            return EventResult::Consumed(None);
+        }
+
+        if let Ok(mut clipboard) = get_clipboard().lock() {
+            *clipboard = Some(ClipboardEntry {
+                path: path.clone(),
+                operation: ClipboardOperation::Copy,
+            });
+        }
+        ctx.editor.set_status(format!("Yanked: {}", path.display()));
```
[the remark this answers is not in the document]
Comment (Dheebz) [reply helix-term/src/ui/file_explorer.rs]: Resolved with https://github.com/helix-editor/helix/pull/15174/commits/ee86bbd96c82ef762e9c75142b93ea93c269672d

File: helix-term/src/ui/file_explorer.rs [hunk 4]
```diff
@@ -0,0 +1,1007 @@
...
+fn copy_dir_recursive(src: &Path, dest: &Path) -> std::io::Result<()> {
+    std::fs::create_dir_all(dest)?;
+    for entry in std::fs::read_dir(src)? {
+        let entry = entry?;
+        let src_path = entry.path();
+        let dest_path = dest.join(entry.file_name());
+        if src_path.is_dir() {
+            copy_dir_recursive(&src_path, &dest_path)?;
+        } else {
+            std::fs::copy(&src_path, &dest_path)?;
```
[the remark this answers is not in the document]
Comment (Dheebz) [reply helix-term/src/ui/file_explorer.rs:958]: Resolved with https://github.com/helix-editor/helix/pull/15174/commits/ee86bbd96c82ef762e9c75142b93ea93c269672d

File: helix-term/src/ui/file_explorer.rs [hunk 5]
```diff
@@ -0,0 +1,1121 @@
...
+
+                let new_path = parent.join(input);
+
+                // Check if destination exists
+                if new_path.exists() {
+                    cx.editor.set_error(format!("'{}' already exists", input));
+                    return;
+                }
+
+                match std::fs::rename(&path, &new_path) {
```
[the remark this answers is not in the document]
Comment (Dheebz) [reply helix-term/src/ui/file_explorer.rs:360]: This is intentional. `../../other_dir/file` is expected behavior
